### DISCUSSION TITLE
Problema con router-view

### DISCUSSION THREAD

**Discussion by Edgarcastrop** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#543] [Q&A]
Buenos dias, disculpen pero cuando descargo y hago el comando "npm run serve" me sale las cosas predefinidas, el problema es cuando entro visual studio code que cuando entro al archivo "App.vue" me marca error en el router view y ya no puedo actualizar la pagina ni volver a correr el "npm run serve" 
PD: con o sin modificaciones pasa

![Captura](https://user-images.githubusercontent.com/99065409/160293427-3f69c5f1-e912-4d2d-b676-349437a23f77.JPG)

**Comment by doguedogue** [comment]
Tienes que ejecutar para que te baje las dependencias antes
`npm install`

Síguete este ejemplo del MC
https://github.com/romarpla/FrontEnd-Mision/tree/main/05%20-%20VUEJS%2Fprogramas%2FVUEX

**Reply by Edgarcastrop** [reply]
Antes o despues de crear la carpeta?

**Comment by GloriaCoralCerecedo** [comment]
Yo lo que hice es que metí el router-view en el nav

**Reply by ChristianDeM** [reply]
tiene que  de  ir dentro de un `<div><div/>` para que no te marque en rojo

**Comment by Dfredude** [comment]
A lo que veo no estas poniendo todos los elementos dentro de un contenedor como tal:
```html
<section>
    elements....
<section/>
```
**_elements..._** tiene que ser todos los elementos que pusiste en _**template**_ incluyendo el routerview. En pocas palabras tienes que poner un elemento que contenga todos los elementos que vas a usar en tu componente. Esto ocurre en todos los frameworks y es algo basico. No quieres tener un despilfarro de componenetes que representan una cantidad variable de elementos. Es por limpieza de codigo. 

Para usar routerview asegurate de tener la dependecia instalada corriendo el comando `npm install vue-router@4` en la carpeta del proyecto. Saludos!
